src/key-value-engine/rocksdb.md

```diff
@@ -0,0 +1,155 @@
+# RocksDB
+
+RocksDB is a persistent key-value store for fast storage environment.
+Here are some highlight features from RocksDB:
+
+1. RocksDB uses a log structured database engine, written entirely in
+   C++, for maximum performance. Keys and values are just
+   arbitrarily-sized byte streams.
+2. RocksDB is optimized for fast, low latency storage such as flash
+   drives and high-speed disk drives. RocksDB exploits the full
+   potential of high read/write rates offered by flash or RAM.
+3. RocksDB is adaptable to different workloads. From database storage
+   engines such as MyRocks to application data caching to embedded
+   workloads, RocksDB can be used for a variety of data needs.
+4. RocksDB provides basic operations such as opening and closing a
+   database, reading and writing to more advanced operations such as
+   merging and compaction filters.
+
+TiKV uses RocksDB because RocksDB is mature and high-performance. In
+this section, we will explore how TiKV uses RocksDB. We won't talk
+about basic features like `Get`, `Put`, `Delete`, and `Iterate` here
+because their usage is simple and clear and works well too. Instead,
+we'll focus some special features used in TiKV below.
+
+## [Prefix Bloom Filter](https://github.com/facebook/rocksdb/wiki/RocksDB-Bloom-Filter)
+
+A Bloom Filter is a magical data structure that uses a little resource
+but helps a lot. We won't explain the whole algorithm here. If you are
+not familiar with Bloom Filters, you can think it as a black box
+inside a dataset, which can tell you if a key *probably* exists or
+*definitely* does not without actually searching the
+dataset. Sometimes Bloom Filter gives you a false-positive answer
+although it rarely happens.
+
+TiKV uses a Bloom Filter as well as a variant which is called Prefix Bloom
+Filter (PBF). Instead of telling you if a whole key exists in a
+dataset or not, PBF tells you if there are some other keys with the
+same prefix exists.
```

Comment:
Can you point out the advantages and disadvantages of PBF, such as much more less memory consume?